modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -201,18 +201,23 @@ def build_contentions(data)
           end
         end
 
-        def get_phase_type_indicator_array(data)
-          return if data[:benefit_claim_details_dto][:phase_type_change_ind].nil?
+        def current_phase_back(data)
+          if data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)&.[](0)&.dig(:phase_type_change_ind).nil?
+            return false
+          end
 
-          data = data[:benefit_claim_details_dto][:phase_type_change_ind]
-          data.split('')
-        end
+          indicator = latest_phase_type_change_indicator(data).split('')
+          return false if indicator == 'N'
 
-        def current_phase_back(data)
-          return false if data[:benefit_claim_details_dto][:phase_type_change_ind].nil?
+          indicator.first.to_i > indicator.last.to_i
+        end
 
-          pt_ind_array = get_phase_type_indicator_array(data)
-          pt_ind_array.first.to_i > pt_ind_array.last.to_i
+        def latest_phase_type_change_indicator(data)
+          if data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status).is_a?(Array)
+            data[:benefit_claim_details_dto][:bnft_claim_lc_status][0][:phase_type_change_ind]
+          elsif data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status, :phase_type)
+            data[:benefit_claim_details_dto][:bnft_claim_lc_status][:phase_type_change_ind]
+          end
```

Comment:
I'd vote to throw this to an array and flatten it again instead of doing this dual check we used before. seems cleaner to me. Also, rails conventions you can use `.first` on arrays. I believe you can use quite a few actually. `.first`, `.second`, `.third`, the possibilities are endless. (that's not true, I'm sure it ends at some point - I just tested to `.fifth` and it worked).

You could essentially replace all this logic with 
```
[data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)].flatten.first&.dig(:phase_type_change_ind)
```
but that starts to get a little messy as well.